explorer/server.go

```diff
@@ -122,47 +122,66 @@ func (a *App) getNode(w http.ResponseWriter, r *http.Request) {
 		log.Warn().Str("Couldn't find entry to redis", string(err.Error())).Msg("")
 
 	}
-	if value != "" {
-		w.Write([]byte(value))
-		return
+	// No value, fetch data from the node
+	if value == "" {
+		nodeInfo, err := a.fetchNodeData(r.Context(), nodeID)
+		if err != nil {
+			log.Error().Err(errors.Wrap(err, "could not fetch node data")).Msg("")
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(http.StatusText(http.StatusInternalServerError)))
+			return
+		}
+		// Save value in redis
+		// caching for 30 mins
+		marshalledInfo, err := json.Marshal(nodeInfo)
+		if err != nil {
+			log.Error().Err(errors.Wrap(err, "could not marshal node info")).Msg("")
+		}
+		err = a.SetRedisKey(fmt.Sprintf("GRID3NODE:%s", nodeID), marshalledInfo, 30*60)
+		if err != nil {
+			log.Fatal().Err(errors.Wrap(err, "Couldn't cache to redis")).Msg("connection error")
```

Comment:
Fatal is going to terminate the full server. i already had this comment on the code. this is extreme. A wraning, or even error level is enough here, the only complication is we recalculate everything next time we do the same request